src/neurostatslib/basis.py

```diff
@@ -845,71 +738,114 @@ def _check_n_basis_min(self) -> None:
         """
         if self._n_basis_funcs < 1:
             raise ValueError(
-                f"Object class {self.__class__.__name__} requires >= 1 basis elements. {self._n_basis_funcs} basis elements specified instead"
+                f"Object class {self.__class__.__name__} requires >= 1 basis elements. "
+                f"{self._n_basis_funcs} basis elements specified instead"
+            )
+
+    def _check_rates(self):
+        """
+        Check if the decay rates list has duplicate entries.
+
+        Raises
+        ------
+        ValueError
+            If two or more decay rates are repeated, which would result in a linearly
+            dependent set of functions for the basis.
+        """
+        if len(set(self._decay_rates)) != len(self._decay_rates):
+            raise ValueError(
+                "Two or more rate are repeated! Repeating rate will result in a "
+                "linearly dependent set of function for the basis."
             )
 
-    def _evaluate(self, *sample_pts: NDArray) -> NDArray:
+    def _check_sample_range(self, sample_pts: NDArray):
         """
-        Generate basis functions with given spacing.
+        Check if the sample points are all positive.
```

Comment:
why is this the only basis that this runs on? don't they all require sample pts to be in some domain?